backend/api/professor/messages.py

```
@@ -0,0 +1,278 @@
+async def get_received_messages(
+    limit: int = 50,
+    offset: int = 0,
+    current_user: User = Depends(require_professor),
+    db: Session = Depends(get_db)
+):
+    """Get messages received by the professor (replies from students)"""
+    
+    # Get messages where the professor is the student (replies to their messages)
+    messages = db.query(ProfessorStudentMessage).filter(
+        ProfessorStudentMessage.student_id == current_user.id,
+        ProfessorStudentMessage.is_reply == True
```

Ruff (0.13.1)

Avoid equality comparisons to `True`; use `ProfessorStudentMessage.is_reply:` for truth checks

Replace with `ProfessorStudentMessage.is_reply`

(E712)